modules/custom/falcon_commerce/src/Plugin/rest/resource/DonationResource.php

```diff
@@ -0,0 +1,294 @@
+
+  /**
+   * User account entity.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  private $account;
+
+  /**
+   * Profile entity.
+   *
+   * @var \Drupal\profile\Entity\ProfileInterface
+   */
+  private $profile;
+
+  /**
+   * Constructs a Drupal\falcon_commerce\Plugin\rest\resource\DonationResource object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param array $serializer_formats
+   *   The available serialization formats.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, array $serializer_formats, LoggerInterface $logger, EntityTypeManagerInterface $entity_type_manager, TimeInterface $time) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $serializer_formats, $logger);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->time = $time;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('falcon_commerce'),
+      $container->get('entity_type.manager'),
+      $container->get('datetime.time')
+    );
+  }
+
+  /**
+   * Handle POST requests.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Request object.
+   *
+   * @return \Drupal\rest\ResourceResponse
+   *   Response object.
+   */
+  public function post(Request $request) {
+    $this->logger->info($request->getContent());
+
+    // TODO: check if we can get this data some other way provided by REST.
+    // Look into the serialization_class.
+    $this->data = json_decode($request->getContent(), TRUE);
+
+    $order = $this->createOrder();
+
+    // TODO: add payment processing.
+    // $this->>processPayment($order);
+
+    $response = new ResourceResponse($order, 200);
+
+    return $response;
+  }
+
+  /**
+   * Creates an order from given data.
+   */
+  protected function createOrder() {
+    $store_ids = [];
+    $order_items = [];
+    $variation_sku = 'donation';
+    /** @var \Drupal\commerce_product\Entity\ProductVariationInterface $variation */
+    $variations = $this->entityTypeManager->getStorage('commerce_product_variation')->loadByProperties(['sku' => $variation_sku]);
+    $variation = reset($variations);
+
+    if (!$store_ids) {
```

Comment:
store_ids will always be empty at this point.

Reply:
True, true...